presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/RowEncoder.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder;
+
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.type.SqlDate;
+import io.prestosql.spi.type.SqlTime;
+import io.prestosql.spi.type.SqlTimeWithTimeZone;
+import io.prestosql.spi.type.SqlTimestamp;
+import io.prestosql.spi.type.SqlTimestampWithTimeZone;
+
+import java.nio.ByteBuffer;
+
+import static io.prestosql.spi.StandardErrorCode.GENERIC_USER_ERROR;
+import static java.lang.String.format;
+
+public interface RowEncoder
+{
+    byte[] toByteArray();
+
+    void clear();
+
+    RowEncoder putNullValue(EncoderColumnHandle columnHandle);
+
+    default RowEncoder put(EncoderColumnHandle columnHandle, Object value)
```

Comment:
please rename the `put` methods to `putObject`, `putLong`, `putInt` ...
Having that many overrides is very error prone. Especially when there are built-in type conversions in the language between the types for which you provide overrides.